### ISSUE TITLE
Prefab brush throw an error

### ISSUE THREAD

**Issue by AndreyChernykh** [Unity-Technologies/2d-extras#64]
Unity **2019.1.0a7**

Hi! I'm trying to use prefab brush in my project as shown in official unity tutorial here -> https://unity3d.com/learn/tutorials/topics/2d-game-creation/custom-tilemap-brush-spawn-prefabs?playlist=17093 but it doesn't work. 

It just places one object and then it doesn't paint more prefabs and there are thrown some exceptions in the console

```
NullReferenceException: Object reference not set to an instance of an object
UnityEditor.PrefabBrush.Paint (UnityEngine.GridLayout grid, UnityEngine.GameObject brushTarget, UnityEngine.Vector3Int position) (at Assets/vendor/2d-extras/Tilemap/Brushes/Prefab Brush/Scripts/Editor/PrefabBrush.cs:20)
UnityEditor.PaintableSceneViewGrid.Paint (UnityEngine.Vector3Int position) (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/PaintableSceneViewGrid.cs:169)
UnityEditor.PaintableGrid.HandleBrushPaintAndErase () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/PaintableGrid.cs:370)
UnityEditor.PaintableGrid.OnGUI () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/PaintableGrid.cs:100)
UnityEditor.PaintableSceneViewGrid.OnSceneGUI (UnityEditor.SceneView sceneView) (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/PaintableSceneViewGrid.cs:81)
UnityEditor.SceneView.CallOnSceneGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:2912)
UnityEditor.SceneView.HandleSelectionAndOnSceneGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:2159)
UnityEditor.SceneView.OnGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:2020)
```

**Comment by ChuanXin-Unity** [comment]
Hi, this will be fixed in a future version of Unity (both for 2018 and 2019 lines)!

**Comment by p1code** [comment]
Unity 2018.3.0f2

i try to modify ‘PrefabBrush.cs’, it seems solved, but can paint prefab under the active tilemap only.

public class PrefabBrush : GridBrushBase
=>
public class PrefabBrush : GridBrush

then

public class PrefabBrushEditor : GridBrushEditorBase
=>
public class PrefabBrushEditor : GridBrushEditor

then

protected void OnEnable()
{
	m_SerializedObject = new SerializedObject(target);
=>
protected override void OnEnable()
{base.OnEnable();
	m_SerializedObject = new SerializedObject(target);

**Comment by DanarKayfi** [comment]
reproduced in 2018.3.0f2

**Comment by Krummelz** [comment]
> Unity 2018.3.0f2
> 
> i try to modify ‘PrefabBrush.cs’, it seems solved, but can not limited to children of the currently selected GameObject.
> 
> public class PrefabBrush : GridBrushBase
> =>
> public class PrefabBrush : GridBrush
> 
> then
> 
> public class PrefabBrushEditor : GridBrushEditorBase
> =>
> public class PrefabBrushEditor : GridBrushEditor
> 
> then
> 
> protected void OnEnable()
> {
> m_SerializedObject = new SerializedObject(target);
> =>
> protected override void OnEnable()
> {base.OnEnable();
> m_SerializedObject = new SerializedObject(target);

This solved it for me! Thanks!

**Comment by bqback** [comment]
> Unity 2018.3.0f2
> 
> i try to modify ‘PrefabBrush.cs’, it seems solved, but can not limited to children of the currently selected GameObject.
> 
> public class PrefabBrush : GridBrushBase
> =>
> public class PrefabBrush : GridBrush
> 
> then
> 
> public class PrefabBrushEditor : GridBrushEditorBase
> =>
> public class PrefabBrushEditor : GridBrushEditor
> 
> then
> 
> protected void OnEnable()
> {
> m_SerializedObject = new SerializedObject(target);
> =>
> protected override void OnEnable()
> {base.OnEnable();
> m_SerializedObject = new SerializedObject(target);

Thank you for this fix (worked on 2018.3.0f2)! Although there's a problem: while the brush no longer throws an error and resets, it now draws a new sprite for every little movement of my mouse cursor within a single tile (so rapidly moving my mouse back and forth across a few pixels creates dozens of the same object), which I doubt to be the intended effect, Same goes for the eraser -- moving the eraser back and forth within a single tile allows me to quickly erase all of those excess sprites.

**Comment by ChuanXin-Unity** [comment]
This has been fixed in Unity 2018.3.4f1, or you can continue to use the PrefabBrush as modified!
